source/common/config/delta_subscription_state.cc

```diff
@@ -113,17 +118,19 @@ void DeltaSubscriptionState::handleBadResponse(const EnvoyException& e, UpdateAc
   ack.error_detail_.set_code(Grpc::Status::GrpcStatus::Internal);
   ack.error_detail_.set_message(e.what());
   disableInitFetchTimeoutTimer();
-  ENVOY_LOG(warn, "delta config for {} rejected: {}", type_url_, e.what());
-  callbacks_.onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::UpdateRejected, &e);
+  ENVOY_LOG(warn, "delta config for {} rejected: {}", type_url(), e.what());
+  callbacks().onConfigUpdateFailed(Envoy::Config::ConfigUpdateFailureReason::UpdateRejected, &e);
```

Comment:
What's the point is `callback() instead of `callback_` here and below? It's not virtual function anyway.

Reply:
That's because `callbacks_` has been factored up to the base class `SubscriptionState`. I, in my less than masterful command of the motivations behind C++ design patterns, actually would prefer to do what you're asking about: making the `callbacks_` variable protected, and directly using it. But, from what I understand, Google/Envoy style says private with protected accessor, so I'm just following that.